### PR TITLE
Add option to eagerly scrub the disk

### PR DESCRIPTION
**Pull request by Shinzu**
### What does this PR do?
adds the option to rewrite the disk with zeros during provisioning or not (which is the default behaviour in vmware)
### What issues does this PR fix or reference?
#44040 
### Previous Behavior
you were not able to choose it disk should be rewrites with zero or not

### New Behavior
option to choose is now available

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

### CODE REVIEW

File: doc/topics/cloud/vmware.rst [hunk 1]
```diff
@@ -283,6 +283,9 @@ Set up an initial profile at ``/etc/salt/cloud.profiles`` or
         thin_provision
             Specifies whether the disk should be thin provisioned or not. Default is ``thin_provision: False``.
             .. versionadded:: 2016.3.0
+        eagerly_scrub
+            Specifies whether the disk should be rewrite with zeros during thick provisioning or not.
+            Default is ``eagerly_scrub: False``.
```
Comment (rallytime) [inline doc/topics/cloud/vmware.rst]: Can you add a `.. versionadded:: Oxygen` tag here?

**Review by rallytime — CHANGES_REQUESTED** [review]
Thank you for adding this option @Shinzu and welcome!

I had one minor request, other than that, this looks good to me.

**Review by gtmanfred — APPROVED** [review]

File: doc/topics/cloud/vmware.rst (hunk 1 above)
Comment (Shinzu) [reply doc/topics/cloud/vmware.rst]: i added the version tag

**Review by cetanu — APPROVED** [review]
eagerlyScrub property is documented here: 

https://code.vmware.com/apis/197/vsphere#/doc/vim.vm.device.VirtualDisk.FlatVer2BackingInfo.html

Looks good to me 👍

File: doc/topics/cloud/vmware.rst (hunk 1 above)
Comment (rallytime) [reply doc/topics/cloud/vmware.rst]: thanks!

**Review by rallytime — APPROVED** [review]

**Pull request merged**